client/src/app/components/employee-settings/employee-settings.component.ts

```diff
@@ -0,0 +1,82 @@
+import {Component, Input, OnInit} from '@angular/core';
+import {ManageMembersService} from '../../services/manage-members.service';
+import {ActivatedRoute} from '@angular/router';
+import {User} from '../../models/user.model';
+
+@Component({
+  selector: 'app-employee-settings',
+  templateUrl: './employee-settings.component.html',
+  styleUrls: ['./employee-settings.component.scss']
+})
+export class EmployeeSettingsComponent implements OnInit {
+  @Input() employee: User;
```

Comment:
So instead of injecting from parent to child ( because later i think we wont have all the informations of the user at once in the table), I think we could just do a api call just to get the info of this particular user, this way the component would be reusable for example for the update profile of the logged in user.

Reply:
I don't understand what you mean. 
What I have implemented is on generation of this component, it will make an API call to get the user information.
The component is entirely reusable, the only consideration we'd have to make is how we pass the ID

Reply:
Since right now, it's passed through the url.
What we could do is just make a route for the settings to pass the ID, which could also work as well, and might make more sense.
I don't know if there's a cleaner way to do this. In React, I would just pass the ID down to the component, and display only that component. However in Angular, it kind of feels cleaner to just display the new page through a different route.

Reply:
Oh sorry I misread, you are right i saw the API call in the ngOnInit(). Okay this is good then. Just another think instead of writing the whole function inside the ngOnInit() to get the particular user, i would just suggest to move it outside in another method called getEmployee() and then call this method inside ngOnInit().

Reply:
I was getting the ID in my other projects through the URL too I think it's fine for now but for sure we can so some reaserch.

Reply:
Hmm, well, I'm not sure how I could decompose it, right?
Like for example, I could refactor out the line of code from the service call + the service subscription functionality, but the service subscription functionality is fairly coupled with the initialization of the state of the component, so I feel like if I removed that part, it wouldn't make much sense.

Or if I decided to remove the line of code from the service call, I would still need to pass in a lambda function to initialize the state, which would end up looking like the same thing, but with an extra method, adding unnecessary code bloat.

In either of these cases, I'm not sure they are the best decisions. Do you have something particular in mind? 
Maybe I can add a comment to make it harder to miss that we are getting an Employee?

Reply:
I just meant like this:
![Screen Shot 2020-10-31 at 4 19 12 PM](https://user-images.githubusercontent.com/35696935/97789192-d0f8ad00-1b94-11eb-9dc8-685fe557f17e.png)

Reply:
Yeah, so that's my first point. Okay, I can do that though. 
I think it actually makes more sense now that I think about it, I was already initializing some variables in the component in other functions. I'll update this soon.

Reply:
Fixed!